### PR TITLE
SDO access hook

### PR DESCRIPTION
**Pull request by leducp**

### CODE REVIEW

File: lib/src/CoE/mailbox/response.cc [hunk 1]
```diff
@@ -212,21 +232,43 @@ namespace kickcat::mailbox::response
...
+            auto* entry = &object->entries.at(subindex);
+            if (not isDownloadAuthorized(entry))
             {
                 abort(CoE::SDO::abort::READ_ONLY_ACCESS);
                 return ProcessingResult::FINALIZE;
             }
 
-            uint32_t entry_size = entry.bitlen / 8;
-            std::memcpy(entry.data, payload_ + 4 + size, entry_size);
+            beforeHooks(entry);
```
Comment (nicolas71640) [inline lib/src/CoE/mailbox/response.cc:242]: How will we know if the master ask for a upload or download ?

Comment (leducp) [reply lib/src/CoE/mailbox/response.cc:242]: No knowledge of the access type at this point: do you think it is needed?

Comment (nicolas71640) [reply lib/src/CoE/mailbox/response.cc:242]: Well, I think this the least we can tell to the caller : what the master trying to do, as these hooks are made to react on these "demands"

Comment (leducp) [reply lib/src/CoE/mailbox/response.cc:242]: Fair enough: I'll add the command to the request

**Review by nicolas71640 — APPROVED** [review]

**Pull request merged**